### PR TITLE
Remove deprecated modules

### PR DESCRIPTION
**Pull request by wvu**
What it says on the tin.
- [x] Verify that the removal dates are correct

Related to #7273.

### CODE REVIEW

File: modules/exploits/windows/smb/psexec_psh.rb [hunk 1]
```diff
@@ -14,9 +14,6 @@ class MetasploitModule < Msf::Exploit::Remote
   # Exploit mixins should be called first
   include Msf::Exploit::Remote::SMB::Client::Psexec
   include Msf::Exploit::Powershell
-  include Msf::Module::Deprecated
```
Comment (wvu-r7) [inline modules/exploits/windows/smb/psexec_psh.rb]: Undeprecating ```psexec_psh```. Please see commit message for details.

**Comment by busterb** [comment]
## Release Notes

The following deprecated modules were removed:
auxiliary/dos/http/android_stock_browser_iframe_dos_cve_2012_6301
auxiliary/gather/apache_karaf_command_execution
auxiliary/gather/dns_bruteforce
auxiliary/gather/dns_cache_scraper
auxiliary/gather/dns_info
auxiliary/gather/dns_reverse_lookup
auxiliary/gather/dns_srv_enum
auxiliary/scanner/netbios/nbname_probe
exploits/linux/http/struts_dmi_exec
post/windows/manage/smart_migrate

The following module was undeprecated:
exploits/windows/smb/psexec_psh

**Comment by busterb** [comment]
Interestingly, I heard that we actually fixed the idiosyncracy between _psh and regular psexec.
It was rooted down to a datastore issue when you saved and restored.

**Comment by wvu-r7** [comment]
Ah, that PR was #7121. That was a weird one. Feel free to redeprecate ```psexec_psh``` in the future once we're sure the discrepancies are ironed out. People need to speak up.

**Pull request merged**